src/manuel/testing.py

```diff
@@ -184,3 +188,45 @@ def TestSuite(m, *paths, **kws):
             suite.addTest(TestCase_class(m, regions, globs, **kws))
 
     return suite
+
+
+_not_word = re.compile(r'\W')
+class TestFactory:
+
+    def __init__(self, m):
+        self.m = m
+
+    def __call__(self, path):
+        base = os.path.dirname(os.path.abspath(
+            sys._getframe(2).f_globals['__file__']
+            ))
+        path = os.path.join(base, path)
+        with open(path) as f:
+            test = f.read()
+
+        m = self.m
+
+        def test_file(self, setup=lambda i: None):
+            if isinstance(self, types.FunctionType):
```

Comment:
My future self requests a comment above this if-statement telling him that this bit of code handles the decorator use case. ;)

Reply:
Good idea. Will do.

Reply:
Done